cmd/admin-handlers.go

```diff
@@ -336,6 +337,132 @@ func (a adminAPIHandlers) PerfInfoHandler(w http.ResponseWriter, r *http.Request
 	return
 }
 
+// LockEntry .......
+type LockEntry struct {
+	Timestamp  time.Time `json:"time"`       // Timestamp set at the time of initialization.
+	Name       string    `json:"name"`       // Name contains info like bucket, object etc
+	Type       string    `json:"type"`       // Bool whether write or read lock.
+	Source     string    `json:"source"`     // Source which created the lock
+	ServerList []string  `json:"serverlist"` // RPC path of servers issuing the lock.
+	Client     string    `json:"client"`     // RPC path of client claiming lock.
+	ID         string    `json:"id"`         // UID to uniquely identify request of client.
+}
+
+func setLockEntry(l lockRequesterInfo, name, server string) *LockEntry {
+	entry := &LockEntry{Timestamp: l.Timestamp, Name: name, ServerList: []string{server}, Client: l.Node, Source: l.Source, ID: l.UID}
+	if l.Writer {
+		entry.Type = "Write"
+	} else {
+		entry.Type = "Read"
+	}
+	return entry
+}
+
+// LockEntries - To sort the locks
+type LockEntries []LockEntry
+
+func (l LockEntries) Len() int {
+	return len(l)
+}
+
+func (l LockEntries) Less(i, j int) bool {
+	return l[i].Timestamp.After(l[j].Timestamp)
+}
+
+func (l LockEntries) Swap(i, j int) {
+	l[i], l[j] = l[j], l[i]
+}
+
+type lockEntryMap map[string]*LockEntry
+
+func sliceLockEntries(entryMap lockEntryMap) LockEntries {
+	var lockEntries LockEntries
+	for _, v := range entryMap {
+		lockEntries = append(lockEntries, *v)
+	}
+	return lockEntries
+}
+
+func serializeLockEntries(entryMap lockEntryMap, server string, locksMap *LocksMap) lockEntryMap {
+	if locksMap != nil {
+		for k, v := range locksMap.LockMap {
+			for _, lockReqInfo := range v {
+				if val, ok := entryMap[lockReqInfo.UID]; ok {
+					val.ServerList = append(val.ServerList, server)
+				} else {
+					entryMap[lockReqInfo.UID] = setLockEntry(lockReqInfo, k, server)
+				}
+			}
+		}
+	}
+	return entryMap
+}
```

Comment:
I think the code will be simpler if you remove the type type lockEntryMap map[string]*LockEntry and combine
sliceLockEntries() serializeLockEntries() and the lines
```
	var entryMap lockEntryMap
	entryMap = make(map[string]*LockEntry)

 	for i := 0; i < len(globalAdminPeers); i++ {
		// Convert to the first map
		entryMap = serializeLockEntries(entryMap, globalAdminPeers[i].addr, &reply[i])
	}

 	// Convert to ListEntries
	locklist := sliceLockEntries(entryMap)
	sort.Sort(locklist)
	if len(locklist) > 10 {
		locklist = locklist[:10]
	}
```

into one function:
```
func topLockEntries(peers adminPeers, lockMaps []LocksMap) LockEntries {
        entryMap := make(map[string]*LockEntry)
        for _, lockMap := range lockMaps {
                if lockMap == nil {
                        continue
                }
                for k, v := range locksMap.LockMap {
                        for _, lockReqInfo := range v {
                                if val, ok := entryMap[lockReqInfo.UID]; ok {
                                        val.ServerList = append(val.ServerList, peers[i].addr)
                                } else {
                                        entryMap[lockReqInfo.UID] = setLockEntry(lockReqInfo, k, peers[i].addr)
                                }
                        }
                }
        }
        var lockEntries LockEntries
        for _, v := range entryMap {
                lockEntries = append(lockEntries, *v)
        }
        sort.Sort(lockEntries)
        sort.Reverse(lockEntries)
        if len(lockEntries) > 10 {
                lockEntries = lockEntries[:10]
        }
        return lockEntries
}
```

Reply:
done